src/hipercow/dide/web.py

```diff
@@ -0,0 +1,286 @@
+        self._client.logout()
+
+    def headnodes(self) -> list[str]:
+        data = {"user": encode64("")}
+        response = self._client.request("POST", "_listheadnodes.php", data=data)
+        return _client_parse_headnodes(response.text)
+
+    def check_access(self) -> None:
+        _client_check_access(self._cluster, self.headnodes())
+
+    def logged_in(self) -> bool:
+        return self._client.logged_in()
+
+    def submit(self, path: str, name: str) -> str:
+        data = _client_body_submit(path, name, self._cluster)
+        response = self._client.request("POST", "submit_1.php", data=data)
+        return _client_parse_submit(response.text)
+
+    def cancel(self, dide_id: str) -> bool:
+        data = _client_body_cancel(dide_id, self._cluster)
+        response = self._client.request("POST", "cancel.php", data=data)
+        return _client_parse_cancel(response.text)
+
+    def log(self, dide_id: str) -> str:
+        data = _client_body_log(dide_id, self._cluster)
+        response = self._client.request("POST", "showjobfail.php", data=data)
+        return _client_parse_log(response.text)
+
+    def status_user(self, state="*") -> list[DideTaskStatus]:
+        data = _client_body_status_user(
+            state, self._client.username(), self._cluster
+        )
+        response = self._client.request("POST", "_listalljobs.php", data=data)
+        return _client_parse_status_user(response.text)
+
+    def status_job(self, dide_id: str) -> TaskStatus:
+        query = _client_query_status_job(dide_id, self._cluster)
+        response = self._client.request("GET", "api/v1/get_job_status/", query)
+        return _client_parse_status_job(response.text)
+
+    def software(self):
+        response = self._client.request(
+            "GET", "api/v1/cluster_software", public=True
+        )
+        return _client_parse_software(response.json())
+
+
+def _client_check_access(cluster: str, valid: list[str]) -> None:
+    if cluster in valid:
+        return
+    if len(valid) == 0:
+        msg = "You do not have access to any cluster"
+    elif len(valid) == 1:
+        msg = f"You do not have access to '{cluster}'; try '{valid[0]}'"
+    else:
+        valid_str = ", ".join(valid)
+        msg = f"You do not have access to '{cluster}'; try one of {valid_str}"
+    raise Exception(msg)
+
+
+def _client_body_submit(path: str, name: str, cluster: str) -> dict:
+    # NOTE: list2cmdline is undocumented but needed.
+    # not documented https://github.com/conan-io/conan/pull/11553/
+    path_call = f"call {list2cmdline([path])}"
+    data = {
+        "cluster": encode64(cluster),
+        "template": encode64("AllNodes"),
+        "jn": encode64(name or ""),  # job name
+        "wd": encode64(""),  # work dir - unset as we do this in the .bat
+        "se": encode64(""),  # stderr
+        "so": encode64(""),  # stdout
+        "jobs": encode64(path_call),
+        "dep": encode64(""),  # dependencies, eventually
```

Comment:
To be clear - this is comma-separated job-ids that this new job will depend on (ie, wait to successfully finish, before starting), rather than any other sorts of dependencies...